packages/client/hmi-client/src/components/home/tera-project-table.vue

```diff
@@ -92,18 +140,46 @@ function formatStatTooltip(stat, displayName) {
 	return `${stat} ${displayName}${stat === 1 ? '' : 's'}`;
 }
 
-function getColumnWidth(columnField: string) {
-	switch (columnField) {
-		case 'description':
-			return 40;
-		case 'name':
-			return 40;
-		case 'score':
-			return 5;
-		default:
-			return 100;
-	}
+async function getProjectAssets(event: PageState = pageState) {
+	pageState = event; // Save the current page state so we still know it when the watch is triggered
+	if (isEmpty(props.searchQuery)) return;
+
+	const { rows, first } = event;
+	const searchQuery = props.searchQuery.toLowerCase().trim();
+
+	// Just fetch the asset data we are seeing in the current page
+	projectsWithKnnMatches.value.slice(first, first + rows).forEach(async (project) => {
+		// If assets were fetched before from when we were on that page don't redo it
+		if (!isEmpty(project.projectAssets) && prevSearchQuery === searchQuery) return;
+
+		const projectWithAssets = (await ProjectService.get(project.id)) as ProjectWithKnnSnippet | null;
+		if (!projectWithAssets) return;
+
+		project.projectAssets = projectWithAssets.projectAssets.filter(
+			(asset) => asset.assetName.toLowerCase().includes(searchQuery) && asset.assetType !== AssetType.Simulation // Simulations don't have names
+		);
+		project.showMore = false;
+		project.snippet = project.description?.toLowerCase().includes(searchQuery)
+			? highlight(project.description, searchQuery)
+			: undefined;
+	});
+
+	prevSearchQuery = searchQuery;
 }
+
+watch(
+	() => props.projects,
+	() => {
+		// Reset the page to 0 when a new search is performed
+		if (pageState.page !== 0) {
+			const firstPageButton = projectTableRef.value.$el.querySelector('.p-paginator-first');
+			firstPageButton?.dispatchEvent(new MouseEvent('click'));
+		}
```

Comment:
This feels awkward, if a new search is performed wouldn't the entire table be refreshed because we have new search results?

Reply:
Oh? then that might be indicative that DataTable has an unexpected reactivity problem/behaviour, and there may be deeper problems that we have not uncovered yet. 

I wonder if using the UUID key trick is a more graceful solution? or whether it will janky? The idea is to force the table to refresh when you have a new search result because of the key-check. But I am not sure if the UI will flicker like mad if we do this.
```
< DataTable :key="tableKey">
</Datatable>


const tableKey = ref(uuid())

function doSearch() {
   ...blahblah...
   tableKey.value = uuid();
} 
```

Reply:
that works